velox/exec/OutputBuffer.cpp

```diff
@@ -358,11 +358,29 @@ void OutputBuffer::addOutputBuffersLocked(int numBuffers) {
   finishedBufferStats_.resize(numBuffers);
 }
 
+void OutputBuffer::updateStatsWithEnqueuedPageLocked(
+    int64_t pageBytes,
+    int64_t pageRows) {
+  bufferedBytes_ += pageBytes;
+  ++bufferedPages_;
+  ++numOutputPages_;
+  numOutputRows_ += pageRows;
+}
+
+void OutputBuffer::updateStatsWithFreedPagesLocked(
+    int numPages,
+    int64_t pageBytes) {
+  bufferedBytes_ -= pageBytes;
+  VELOX_CHECK_GE(bufferedBytes_, 0);
+  bufferedPages_ -= numPages;
+  VELOX_CHECK_GE(bufferedPages_, 0);
+}
+
 bool OutputBuffer::enqueue(
     int destination,
-    std::unique_ptr<SerializedPage> data,
+    std::unique_ptr<SerializedPage> page,
```

Comment:
Meng, these refactorings are nice, but they may it very hard to review the actual changes. Would you split extract these into a separate PR?

Reply:
Reverted.